### ISSUE TITLE
pnpm get stuck when an extra install/postinstall script should also be installed

### ISSUE THREAD

**Issue by Murtatrxx** [pnpm/pnpm#5706]
### Actual behavior: waited for more than 30 mins, but no good luck. It just shows the message
![image](https://user-images.githubusercontent.com/77380166/204141443-315ca88f-5fb6-41d5-ae59-dd0056c59cad.png)
(it failed because i did CTRL C)

### Additional information: please correct me if i'm doing something wrong, i'm new to pnpm

 - `node -v` prints: 18.2
 - Windows, macOS, or Linux?: win11

**Comment by Murtatrxx** [comment]
Probably related with #5909

**Comment by zkochan** [comment]
Do you use `node-linker=hoisted`?

**Comment by Murtatrxx** [comment]
Can confirm this issues also happens with yarn, in yarn on it's getting stuck when linking dependencies, and when you try to start it strangely restarts the console

**Comment by zkochan** [comment]
does it only happen with prisma? does it work if you run `pnpm install --ignore-scripts`?

**Comment by Murtatrxx** [comment]
1, can confirm it happens with all dependencies that has install or postinstall script
2, will try soon probably will work but we want that scripts to be runned

**Comment by Murtatrxx** [comment]
Hmmmm, I think I found the solution. It does something to do with [Windows Legacy Console mode.](https://learn.microsoft.com/en-us/windows/console/legacymode) And I come up with that solution.
1. WIN R (Run) then regedit -- Open registry editor
2. Navigate to Computer/HKEY_CURRENT_USER/Console
3. Double click the option "ForceV2" DWORD parameter then click edit.
4. Swap the value data (0 to 1, 1 to 0)

That fixed my solution.

**Comment by weyert** [comment]
> Hmmmm, I think I found the solution. It does something to do with [Windows Legacy Console mode.](https://learn.microsoft.com/en-us/windows/console/legacymode) And I come up with that solution.

Fascinating, so you are disabling this mode?

**Comment by Murtatrxx** [comment]
Yeah, looks like it. I don't see a single relation between this but...

**Comment by zkochan** [comment]
We have a special command for detecting issues like this: `pnpm doctor`.

Do you think we could detect if legacy mode is on and print an error about it?

**Comment by bchilcott** [comment]
I've just tried disabling legacy console mode, to no avail. Running pnpm install gets stuck on a postinstall step, and after Ctrl + C it shows that windows message. Running any package.json script just immediately shows the windows message too.

![image](https://user-images.githubusercontent.com/38051283/224566385-164b58aa-26b3-4e8d-9860-cbe948261ee9.png)

**Comment by bchilcott** [comment]
> Hmmmm, I think I found the solution. It does something to do with [Windows Legacy Console mode.](https://learn.microsoft.com/en-us/windows/console/legacymode) And I come up with that solution.

Aside from it not working for me, I don't think disabling this option is a viable solution, only a workaround. It seems to prevent Windows Terminal from becoming the default terminal application which obviously isn't great.

**Comment by Murtatrxx** [comment]
🤷‍♂️

**Comment by ChristophWolfPCG** [comment]
still happening, using npm now, that works

**Comment by nicolasembleton** [comment]
I have had this issue as well since 2 days and after some debugging (removing PNPM and node and reinstalling them globally) I managed to pass the package that was blocking, to find out a package-level error saying this:

```
{"time":1693816829686,"hostname":"n.local","pid":28272,"level":"debug","name":"pnpm:link","target":"[...]/node_modules/.pnpm/readable-stream@3.6.2/node_modules/readable-stream","link":"[...]/node_modules/.pnpm/keccak@3.0.3/node_modules/readable-stream"}
{"time":1693816830834,"hostname":"n.local","pid":28272,"level":"debug","name":"pnpm:fetching-progress","attempt":2,"packageId":"registry.npmjs.org/fix/0.0.3","size":22144,"status":"started"}
{"time":1693816831105,"hostname":"n.local","pid":28272,"level":"debug","name":"pnpm:request-retry","attempt":2,"error":{"code":"ERR_PNPM_TARBALL_EXTRACT","attempts":2,"resource":"https://registry.npmjs.org/fix/-/fix-0.0.3.tgz"},"maxRetries":2,"method":"GET","timeout":60000,"url":"https://registry.npmjs.org/fix/-/fix-0.0.3.tgz"}
```

The issue I am now seeing is similar to this one https://github.com/falconair/nodefix/issues/43 and seems to be related to a malformed JSON file (it would seem that there's a comma too many here: 
```
  "directories": {
    "handlers": "./handlers",
  },
```

This ultimately fails saying this:

```
{"time":1693816891477,"hostname":"n.local","pid":28272,"level":"debug","name":"pnpm:fetching-progress","attempt":3,"packageId":"registry.npmjs.org/fix/0.0.3","size":22144,"status":"started"}
{"time":1693816891490,"hostname":"n.local","pid":28272,"level":"warn","name":"pnpm:package-requester","message":"Fetching registry.npmjs.org/fix/0.0.3 failed!","prefix":"[redacted path]"}
{"time":1693816891491,"hostname":"n.local","pid":28272,"level":"error","name":"pnpm","code":"ERR_PNPM_TARBALL_EXTRACT","attempts":3,"resource":"https://registry.npmjs.org/fix/-/fix-0.0.3.tgz","name":"pnpm","err":{"name":"pnpm","message":"Failed to unpack the tarball from \"https://registry.npmjs.org/fix/-/fix-0.0.3.tgz\": SyntaxError: Expected double-quoted property name in JSON at position 237","code":"ERR_PNPM_TARBALL_EXTRACT","stack":"pnpm: Failed to unpack the tarball from \"https://registry.npmjs.org/fix/-/fix-0.0.3.tgz\": SyntaxError: Expected double-quoted property name in JSON at position 237\n    at Worker.<anonymous> ([...]/.cache/node/corepack/pnpm/8.7.1/dist/pnpm.cjs:81781:20)\n    at Object.onceWrapper (node:events:629:26)\n    at Worker.emit (node:events:514:28)\n    at MessagePort.<anonymous> (node:internal/worker:263:53)\n    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:778:20)\n    at exports.emitMessage (node:internal/per_context/messageport:23:28)"}}
```

pnpm: v8.7.1
node: v20.5.1 (it seems I could not "see" the error before and I was using node v18.17.1)

**Comment by zkochan** [comment]
Closing for lack of a reproduction on a supported version.

An install that hung on pnpm 7.17 when an extra install script was present. Build scripts are opt-in and gated now, and a failing one reports its output and exits.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
